Barkane/Assets/Scripts/Test/SceneManager.cs

```diff
@@ -0,0 +1,112 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Analytics;
+
+[ExecuteInEditMode]
+public class SceneManager : MonoBehaviour
+{
+    [SerializeField] private PaperSquares squares;
+    public PaperSquares Squares => squares;
+    [SerializeField] private PaperSqaure copyFrom;
+
+    private BoxCollider meshCollider;
+
+    [SerializeField] private int axisPos = -1;
+    [SerializeField] Orientation orientation;
+
+    public bool GetPlanePosition(Ray mouseRay, out Vector3 hitPoint)
+    {
+        // Bit mask for layer 6 (paper layer)
+        int paperMask = 1 << 6;
```

Comment:
The paper layer should probably only be used for the squares themselves, can we have the plane be its own layer? Layermask for plane position detection would still prob need to also include paper layer for when you click on them, but this way it makes things a little more consistent.